OpenChange/iCalEvent+MAPIStore.m

```diff
@@ -371,106 +472,24 @@ - (void) updateFromMAPIProperties: (NSDictionary *) properties
   value = [properties objectForKey: @"recipients"];
   if (value)
     {
-      NSArray *recipients;
       NSDictionary *dict;
-      NSString *orgEmail, *sentBy, *attEmail;
+      NSString *orgEmail, *sentBy;
       iCalPerson *person;
       iCalPersonPartStat newPartStat;
-      NSNumber *flags, *trackStatus;
-      int i, effective;
+      int effective;
       BOOL organizerIsSet = NO;
 
       [self setOrganizer: nil];
       [self removeAllAttendees];
 
-      recipients = [value objectForKey: @"to"];
-      effective = 0;
-      for (i = 0; i < [recipients count]; i++)
-        {
-          dict = [recipients objectAtIndex: i];
-          person = [iCalPerson new];
-          [person setCn: [dict objectForKey: @"fullName"]];
-          attEmail = [dict objectForKey: @"email"];
-          [person setEmail: attEmail];
- 
-          flags = [dict objectForKey: MAPIPropertyKey (PR_RECIPIENT_FLAGS)];
-          if (!flags)
-            {
-              [self logWithFormat:
-                      @"no recipient flags specified: skipping recipient"];
-              continue;
-            }
+      effective = [self _updateFromAttendeeMAPIProperties: [value objectForKey: @"to"]
+                                                 withRole: @"REQ-PARTICIPANT"
+                                                 outParam: &organizerIsSet];
+      effective += [self _updateFromAttendeeMAPIProperties: [value objectForKey: @"cc"]
+                                                 withRole: @"OPT-PARTICIPANT"
+                                                 outParam: &organizerIsSet];
```

Comment:
I want to make sure it follows the spec to avoid testing empirically with all supported Outlook version. A search in MS-OXOCAL Section 2.2.4.10.7 says the recipient type is 0x01 as required and 0x02 as optional. Searching for Recipient Type property in a Recipient Row shows, at least, two results, they were found MS-OXOMSG Section 2.2.3.1 and MS-OXCMSG 2.2.3.1.2 describing MAPI_TO as 0x01 and MAPI_CC as 0x02. So it is relevant to put this information in both places (commits and code documentation) to avoid the feeling of hack as the whole and exact information is in place.